modules/cog-fdo-shell.c

```diff
@@ -30,6 +44,322 @@ typedef struct {
     CogShell parent;
 } CogFdoShell;
 
+struct wpe_input_touch_event_raw touch_points[10];
+
+extern PwlData wl_data;
```

Comment:
These structures are going to be shareable. All contains structs relative to the keyboard, window and EGL things show  all should be shared independent on the module which use those.

Reply:
I think they should be encapsulated in some singleton class for cleanliness, but just my opinion.